sharball-release.sh

```diff
@@ -0,0 +1,165 @@
+#!/bin/bash
```

Comment:
@XuehaiPan appreciate the work here but to save you effort: I'd really like to see how far we can go with an existing tool e.g. CPack or other rather than creating/maintaining a script like this by hand.

Reply:
@MikeMcQuaid I investigated CPack packaging tool last weekend. But I don't find a way to generate sharball packages using CPack.

[Cpack Generators](https://cmake.org/cmake/help/latest/manual/cpack-generators.7.html#id1):

- [CPack Archive Generator](https://cmake.org/cmake/help/latest/cpack_gen/archive.html)
- [CPack Bundle Generator](https://cmake.org/cmake/help/latest/cpack_gen/bundle.html)
- [CPack Cygwin Generator](https://cmake.org/cmake/help/latest/cpack_gen/cygwin.html)
- [CPack DEB Generator](https://cmake.org/cmake/help/latest/cpack_gen/deb.html)
- [CPack DragNDrop Generator](https://cmake.org/cmake/help/latest/cpack_gen/dmg.html)
- [CPack External Generator](https://cmake.org/cmake/help/latest/cpack_gen/external.html)
- [CPack FreeBSD Generator](https://cmake.org/cmake/help/latest/cpack_gen/freebsd.html)
- [CPack IFW Generator](https://cmake.org/cmake/help/latest/cpack_gen/ifw.html)
- [CPack NSIS Generator](https://cmake.org/cmake/help/latest/cpack_gen/nsis.html)
- [CPack NuGet Generator](https://cmake.org/cmake/help/latest/cpack_gen/nuget.html)
- [CPack PackageMaker Generator](https://cmake.org/cmake/help/latest/cpack_gen/packagemaker.html)
- [CPack productbuild Generator](https://cmake.org/cmake/help/latest/cpack_gen/productbuild.html)
- [CPack RPM Generator](https://cmake.org/cmake/help/latest/cpack_gen/rpm.html)
- [CPack WIX Generator](https://cmake.org/cmake/help/latest/cpack_gen/wix.html)

The most useful generators we are interested in are:

- [CPack Archive Generator](https://cmake.org/cmake/help/latest/cpack_gen/archive.html)
  - 7Z - 7zip - (.7z)
  - TBZ2 (.tar.bz2)
  - TGZ (.tar.gz)
  - TXZ (.tar.xz)
  - TZ (.tar.Z)
  - TZST (.tar.zst)
  - ZIP (.zip)

- [CPack DEB Generator](https://cmake.org/cmake/help/latest/cpack_gen/deb.html)
  - DEB (.deb) for Debian / Ubuntu

- [CPack DragNDrop Generator](https://cmake.org/cmake/help/latest/cpack_gen/dmg.html)
  - DragNDrop (.dmg) for macOS

- [CPack FreeBSD Generator](https://cmake.org/cmake/help/latest/cpack_gen/freebsd.html)
  - FreeBSD (.pkg) for macOS

- [CPack RPM Generator](https://cmake.org/cmake/help/latest/cpack_gen/rpm.html)
  - RPM (.rpm) for Red-Hat / CentOS

Reply:
My mistake, I'm sure this used to be supported but probably has been removed! 

Is there any other software that can be used to generate something like this?